### ISSUE TITLE
[linux] Request for a guide for the steam deck

### ISSUE THREAD

**Issue by EinRainerZufall** [cheatfreak47/NMSDecompressinator#11]
If I have understood it correctly, the script also works on Linux. If so, it would be great for people who don't know much about Linux, like me, that there would be a step by step guide for the Steam Deck.

My main PC has an R9 3900X and it ran ok with occasional stutters on medium graphics settings, and uncompressed I can play it on Ultra with no stutters. So you could probably get more battery life out of the Steam Deck.

**Comment by cheatfreak47** [comment]
@Inovvia any insight here? You're the author of the linux script.

Anyway, the best option for Steam Deck users at the moment is probably to just run the script on your main PC and copy the files over to your Steam Deck, afaik the current linux script hasn't been tested on Arch (which is what Steam OS is based on I think)

**Comment by Inovvia** [comment]
I might need to modify the script to support this version of wine - https://flathub.org/apps/org.winehq.Wine. I'll just update it later and try if it works. Since this seems to be the easiest way for non technical users.

**Comment by Inovvia** [comment]
This might be harder than I thought since steamdeck cannot easily install wine.

**Comment by EinRainerZufall** [comment]
> @Inovvia any insight here? You're the author of the linux script.
> 
> Anyway, the best option for Steam Deck users at the moment is probably to just run the release batch script on your main PC and copy the files over to your Steam Deck, afaik the current linux script hasn't been tested on Arch (which is what Steam OS is based on I think)

I have tried it, but there is always the problem that some objects are invisible in the game. As a precaution, I decompressed and transferred it twice. However, every time I started the game, other objects were invisible.

**Comment by cheatfreak47** [comment]
Weird. That usually happens when people run the script on a library folder that is too deep in the disk or when they don't have enough free space left.

Mind posting your BankSignitures.bin file? (You can zip it and directly drop it on a comment on this thread)

**Comment by Inovvia** [comment]
There's also a new update that might be causing issues for you. Your repacked files from windows might be mismatched if your steam deck is updated. Try running the linux version of the script and report back to us if there are any issues.

**Comment by EinRainerZufall** [comment]
>Mind posting your BankSignitures.bin file? (You can zip it and directly drop it on a comment on this thread)

Here is the [BankSignatures](https://github.com/user-attachments/files/16673640/BankSignatures.zip) file but this is from the latest update and I haven't tried it on the Steam deck yet

>There's also a new update that might be causing issues for you. Your repacked files from windows might be mismatched if your steam deck is updated. Try running the linux version of the script and report back to us if there are any issues.

I quickly tested it today but couldn't get wine to run, I would test it again tomorrow or the day after tomorrow as I have more time.
[console-output_1.txt](https://github.com/user-attachments/files/16674533/console-output_1.txt)

**Comment by Inovvia** [comment]
Did you use the new script following the guide here?  - https://github.com/cheatfreak47/NMSDecompressinator/pull/12 Since i noticed that it did not use the flatpak version from your logs

**Comment by EinRainerZufall** [comment]
Oh... I completely overlooked that, I’ll test it when I get back home.

**Comment by EinRainerZufall** [comment]
@Inovvia 
I have been running the script for about 8 hours now, and for the last 4 hours the console has not moved. What exactly do you need now? Here are the [log.txt](https://github.com/user-attachments/files/16687732/log.txt) and the [BankSignatures.zip](https://github.com/user-attachments/files/16687743/BankSignatures.zip)
<br/><br/>
@cheatfreak47 
> > Mind posting your BankSignitures.bin file? (You can zip it and directly drop it on a comment on this thread)
> 
> Here is the [BankSignatures](https://github.com/user-attachments/files/16673640/BankSignatures.zip) file but this is from the latest update and I haven't tried it on the Steam deck yet
> 
> > There's also a new update that might be causing issues for you. Your repacked files from windows might be mismatched if your steam deck is updated. Try running the linux version of the script and report back to us if there are any issues.
> 
> I quickly tested it today but couldn't get wine to run, I would test it again tomorrow or the day after tomorrow as I have more time. [console-output_1.txt](https://github.com/user-attachments/files/16674533/console-output_1.txt)

I tested it on the Steam Deck with the data of this BankSignature and had again a half invisible infected factory and the control monitor of the landing pad in my own base was invisible, but I forgot to take screenshots.

**Comment by cheatfreak47** [comment]
> half invisible infected factory
> control monitor of the landing pad
this is actually a vanilla bug, sometimes this happens where buildings cull incorrectly or when on low graphics settings

**Comment by EinRainerZufall** [comment]
ok then it will probably not have anything to do with the issue

**Comment by cheatfreak47** [comment]
> I have been running the script for about 8 hours now, and for the last 4 hours the console has not moved. What exactly do you need now? Here are the [log.txt](https://github.com/user-attachments/files/16687732/log.txt) and the [BankSignatures.zip](https://github.com/user-attachments/files/16687743/BankSignatures.zip)**

I can say for sure the script should not take 8 hours to run, based on your log file you dropped here it's made it through some of the files all the way up to `NMSARC.TexAtlas.pak` before it seems to have stopped/when you dumped the log here, and it started running into issues with writing more files, I suspect the problem here is that it's using a Home Directory for storing the unpacked files and there's probably insuffecient space on the disk for this unfortunately.

You may want to delete the folder `nms_working_folder` from `\home\deck` on your Steam Deck, since it probably is using a bunch of space since it seems like the script failed at that point.

To be clear here, this script is still only *officially* supported on Windows, all Linux support for it is still under development, so issues like this are pretty typical if you intend on trying the linux script or whatever.

Anyway, it sounds like you may need to completely reinstall No Man's Sky from scratch on both your Steam Deck and on your PC, and run the Windows script with the fresh copy, and then copy the files over the reinstalled copy on Steam Deck **only after** running the newly installed copy on Deck at least once, to make sure it doesn't run a first run re-validation or whatever.

**Comment by EinRainerZufall** [comment]
@Inovvia 
Here is a quick overview of the file system in case it helps
[files.txt](https://github.com/user-attachments/files/16688141/files.txt)

**Comment by cheatfreak47** [comment]
I appreciate your helping with testing the Linux script, hopefully Inovvia can sort it out- I plan on testing and working on it myself eventually but I don't have as much free time lately, since I recently started a part time job. 

Maybe with the money from that job I should pick up a Steam Deck eventually lol

**Comment by Inovvia** [comment]
I agree with @cheatfreak47. It might be better to start from scratch since the files have already been modified and might cause more issues if you try to rerun the script again. It might be better to run it on windows and copy for now (What I initially did before creating the linux script so I can test for issues).

**Comment by Inovvia** [comment]
I might even try live booting or using steamos in a VM just to test it.

**Comment by EinRainerZufall** [comment]
>I can say for sure the script should not take 8 hours to run, ...

I had it running overnight to be on the safe side

>I suspect the problem here is that it's using a Home Directory for storing the unpacked files and there's probably insuffecient space on the disk for this unfortunately.

I still have about 250gb free so it shouldn't be a problem of free memory

>Anyway, it sounds like you may need to completely reinstall No Man's Sky from scratch on both your Steam Deck and on your PC, and run the Windows script with the fresh copy, and then copy the files over the reinstalled copy on Steam Deck only after running the newly installed copy on Deck at least once, to make sure it doesn't run a first run re-validation or whatever.

Between each try, I always completely deinstalled and reinstalled NMS

**Comment by EinRainerZufall** [comment]
> I might even try live booting or using steamos in a VM just to test it.

SteamOS 2 is based on Debian and can run on any computer, SteamOS 3 is based on Arch and only runs on the Steam Deck. So this might not be super useful
